ee/clickhouse/models/event.py

```diff
@@ -59,28 +61,24 @@ def create_event(
 
     p.produce_proto(sql=INSERT_EVENT_SQL, topic=KAFKA_EVENTS, data=pb_event)
 
-    if team.slack_incoming_webhook or team.organization.is_feature_available("zapier"):
-        # Do a little bit of pre-filtering
-        if event in ActionStep.objects.filter(action__team_id=team.pk, action__post_to_slack=True).values_list(
-            "event", flat=True
-        ):
-            try:
-                celery.current_app.send_task(
-                    "ee.tasks.webhooks_ee.post_event_to_webhook_ee",
-                    (
-                        {
-                            "event": event,
-                            "properties": properties,
-                            "distinct_id": distinct_id,
-                            "timestamp": timestamp,
-                            "elements_list": elements,
-                        },
-                        team.pk,
-                        site_url,
-                    ),
-                )
-            except:
-                pass
+    if Action.objects.filter(team=team).exists():
```

Comment:
This turns out to be quite an expensive query in production, at least for me locally.

```python
In [1]: Action.objects.filter(team_id=2).exists()

SELECT (1) AS "a"
  FROM "posthog_action"
 WHERE "posthog_action"."team_id" = 2
 LIMIT 1

Execution time: 1.480829s [Database: default]
```

Some of it is my local env since the EXPLAIN looks good:

```
EXPLAIN ANALYZE SELECT (1) AS "a"
   FROM "posthog_action"
  WHERE "posthog_action"."team_id" = 2
  LIMIT 1
 
+------------------------------------------------------------------------------------------------------------------------------->
| QUERY PLAN                                                                                                                    >
|------------------------------------------------------------------------------------------------------------------------------->
| Limit  (cost=0.06..0.23 rows=1 width=4) (actual time=0.091..0.092 rows=1 loops=1)                                             >
|   ->  Index Only Scan using posthog_action_team_id_3a21e3a6 on posthog_action  (cost=0.06..34.31 rows=202 width=4) (actual tim>
|         Index Cond: (team_id = 2)                                                                                             >
|         Heap Fetches: 1                                                                                                       >
| Planning Time: 0.085 ms                                                                                                       >
| Execution Time: 0.107 ms                                                                                                      >
+------------------------------------------------------------------------------------------------------------------------------->
EXPLAIN
Time: 0.988s
```

Not sure what the best action here is. I see two points of risk:
1. Query being slow/causing a lot of load. We could mitigate this with extracting the timestamp caching logic into a decorator and using it here.
2. We overload celery due to a lot of extra tasks

I think it makes sense to merge this and keep a very keen eye on production metrics + notify the team. If anything goes south, revert and investigate?

Reply:
Weird 🤔 but things should be alright I think